src/Catel.Test/Catel.Test.Shared/Services/GuidObjectIdGeneratorFacts.cs

```diff
@@ -0,0 +1,42 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="GuidObjectIdGeneratorFacts.cs" company="Catel development team">
+//   Copyright (c) 2008 - 2018 Catel development team. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Catel.Test.Services
+{
+    using System;
+
+    using Catel.Services;
+    using Catel.Test.ViewModels;
+
+    using NUnit.Framework;
+
+    public class GuidObjectIdGeneratorFacts
+    {
+        [TestFixture]
+        public class The_GetUniqueIdentifier_Method
+        {
+            [Test]
+            public void Returns_New_UniqueIdentifier_Even_If_Are_Generated_By_Different_Instances()
```

Comment:
Please add a new test where we use 2 different view models so ensure that they still don't generate the same id (e.g. MyViewModel and PersonViewModel). I believe this is required by some managers for now.

Reply:
Fixed